docs/proposals/20220411-cluster-api-state-metrics.md

```diff
@@ -320,45 +329,51 @@ Common labels:
-| `capi_machinehealthcheck_status_condition`            | `.status.conditions==<condition>` | Gauge | `condition=<condition>`<br>`status=<true\|false>` |        |
-| `capi_machinehealthcheck_status_expected_machines`    | `.status.expectedMachines`        | Gauge |                                                   |        |
-| `capi_machinehealthcheck_status_current_healthy`      | `.status.currentHealthy`          | Gauge |                                                   |        |
-| `capi_machinehealthcheck_status_remediations_allowed` | `.status.remediationsAllowed`     | Gauge |                                                   |        |
+| metric name                                           | value                             | type  | additional labels/tags                                 | xref         |
+|-------------------------------------------------------|-----------------------------------|-------|--------------------------------------------------------|--------------|
+| `capi_machinehealthcheck_created`                     | `.metadata.creationTimestamp`     | Gauge |                                                        | common       |
+| `capi_machinehealthcheck_annotation_paused` *         | `1`                               | Gauge | `paused_value=<paused annotation value>`               |              |
+| `capi_machinehealthcheck_owner`                       | `1`                               | Gauge | `owner_kind=<owner kind>`<br>`owner_name=<owner name>` | [ReplicaSet] |
+| `capi_machinehealthcheck_status_condition`            | `.status.conditions==<condition>` | Gauge | `condition=<condition>`<br>`status=<true\|false>`      |              |
+| `capi_machinehealthcheck_status_expected_machines`    | `.status.expectedMachines`        | Gauge |                                                        |              |
+| `capi_machinehealthcheck_status_current_healthy`      | `.status.currentHealthy`          | Gauge |                                                        |              |
+| `capi_machinehealthcheck_status_remediations_allowed` | `.status.remediationsAllowed`     | Gauge |                                                        |              |
+
+*: A metric will only be exposed if the annotation existst. If so it will always have a value of `1` and expose a label which contains its value. Prometheus would drop labels having an empty value, which is why an empty value would be equal to a not set annotation otherwise.
 
 ### Gaduation Criteria
 
-The initial plan is to add cluster-api-state-metrics as an experimental feature under `exp/` and allow it to be enabled via `tilt`.
+The initial plan is to add kube-state-metrics as to the `./hack/observability` directory and allow it to be enabled via `tilt`.
 
 ## Implementation History
 
+- [x] 09/07/2022: Updated proposal to match current implementation state, removed `_labels` metrics due to lack of functionality in kube-state-metrics
```

Comment:
Is _labels something that could be interesting in the future?

Afaik they support it for node_labels and it can be very very useful

Reply:
A generic `_labels` metric which also re-uses the same flags as for `kube_*_labels` metrics is currently not possible.

However, it is possible to configure your custom `Info` metric which behaves similar but requires an explicit listing of which labels to expose.
To not risk high cardinality and due to the lack of support I removed it for now.

Don't know if we could highlight this in a better way. Maybe at future work?

Reply:
Ah good to know that it can be done via `Info`. Could be very useful to expose some well-known labels that we know are there.

I would say the current state is fine and we can eventually document it when we get to documenting "how to generate your metrics config". Let's just resolve for now

Reply:
But it's an interesting use case for the metric gen tool as it's not super straight-forward

Reply:
I think its more a feature on ksm side / enable a `_labels` metric as for any `kube_*_labels` metric, but just for custom resource.